### DISCUSSION TITLE
Help needed: Using deffered scheduling feature

### DISCUSSION THREAD

**Discussion by MarkoBox** [aleksul/repid#80] [Q&A] [answered]
Hello, Thanks for the library :)

I'm trying to create recuring job which would execute every N days. Looking at the docs I found this:
https://repid.aleksul.space/1.0.0rc3/user_guide/deferred_scheduling/

Example code i had in mind is:

```from datetime import timedelta

from repid import Job
from repid.router import Router, RouterDefaults
import settings

router = Router(defaults=RouterDefaults(queue=settings.rmq_name))


@router.actor
async def delete_stuff_in_db():
    print("delete_stuff_in_db")


def enqueue_delete_stuff_in_db():
    await Job(
        "delete_stuff_in_db",
        settings.rmq_name,
        timeout=timedelta(seconds=30),
        deferred_by=timedelta(days=7)
    ).enqueue()


if __name__ == "__main__":
    enqueue_delete_stuff_in_db()
```

Would running this on single container start suffice, or is there something more I'm missing? 
In case there are several instances of app running how would this behave. Would i need to make sure only one is running it?

Thanks in advance for help :)

**Comment by aleksul** [comment, marked as answer]
Hi 👋 

> or is there something more I'm missing?

Unfortunately, your example won't run as it is. To enqueue a Job, you will need to fix 2 things: 

1. Make `enqueue_delete_stuff_in_db` function async, otherwise you won't be able to do `await`
2. Open a connection, as described in https://repid.aleksul.space/1.0.0rc3/user_guide/initial_setup/

> In case there are several instances of app running how would this behave. Would i need to make sure only one is running it?

Short answer: it depends. In an ideal world, you would set an id to your Job, which will make it unique. This behavior is described [here](https://repid.aleksul.space/1.0.0rc3/user_guide/jobs_actors_and_workers/#job-id-and-uniqueness).

However, not all brokers are supporting this functionality (e.g. current Redis implementation does support it, but RabbitMQ - doesn't). You can, of course, create your own synchronization primitive on top of it.
